Users/justin.pumford/update tools

jpumford opened the pull request:
Enjoy!

Commit:
Add a route to the local proxy for POST requests

There is now an endpoint to get data with a POST rather than a GET
with query parameters. This commit allows us to use this endpoint
while developing locally.

I am having trouble getting the POST endpoint to work correctly and
consistently, so I don't want to merge this yet until that is
resolved.

Commit:
fix(init-repo): replace `domo upload` with `npm run upload`

The instructions included after the setup script is run were not in sync with the documentation.
Change to `npm run upload`

Commit:
fix(readme): add domo login to `getting started`

andrewjensen committed:
Change naming and links to reference public starter-kit

chris-clm09 committed:
Update README.md

* added Prerequisites
* fixed lint doc

andrewjensen committed:
Merge pull request #1 from DomoApps/chris-clm09-patch-1

Update README.md

andrewjensen committed:
Update README.md

- Correct `setup` script
- Add link to component-based-development

chris-clm09 committed:
Merge pull request #2 from DomoApps/andrewjensen-patch-1

Update README.md

andrewjensen committed:
Update README.md

chris-clm09 committed:
Merge pull request #3 from DomoApps/andrewjensen-patch-1

Update README.md

Commit:
Fix(proxy): Proxy now runs locally for post

Fixed all eslint errors

Changed request from data to body property

Commit:
format ternary

Commit:
include body-parser node module in package.json

Commit:
feat(): Add ryuu.js (domo.js) to package.json (#80)

Commit:
Add replaceLicense step in setup script, add banner comment to bundles (#81)

Commit:
chore(): Add AppTeam6 info to package.json (#74)

Commit:
Rename all mobile to responsive

All results of mobile in comments, file names, file references updated
to refer to view as responsive

Commit:
forgot these mobile name change files

Commit:
Fix(switcher): Allow dev to decide tablet view

When an app has two distinct designs the switcher.js gets called. Now
the dev decides what design will be loaded for tablets.

Commit:
update variable nameing and default

Commit:
Fix(switcher): use requestAnimationFrame

No longer depends on a timeout which could cause race conditions. Use
requestAnimationFrame for more consistent behavior and better
performance.

Commit:
Feat(switcher): Correct output per device

Fixed all devices return correct build files. Updated media queries to
cover full range of sizes and devices.

Commit:
remove unnessary testing files

Commit:
Most common default state set

Commit:
Fix: non device based queries

Commit:
Update(README)

Commit:
Fix(switcher): more strict tablet portrait mode

Commit:
fix(): Update da-plop dependency for route platform changes

andrewjensen committed:
chore(): Create framework-specific changelog and contributing docs

Commit:
feat(): Add ngRedux and created common store directories
DAS-4096

sldavidson committed:
Update index.js

Commit:
fix(constants): correct linter error on double declaration of 'constants'

Commit:
feat(app): remove daEvents service as it is replace by Redux. DAS-4096

Commit:
feat(): created test entry file DAS-4146

Commit:
chore(package.json): add cdn dependencies property

Commit:
refactor(desktop&responsive.html): replace hardcoded script tags with generated ones

Commit:
refactor(karma.conf): move cdns to package.json in karma config file

Commit:
refactor(): changed Technology list

jpumford committed:
Merge pull request #6 from DomoApps/refactor/DAS-4173-cdn-injection

CDN Dependencies

Commit:
refactor: rewrite karma.conf to include only necessary test files

Commit:
fix(package.json): update version of angular to 1.5.3

somehow a really old version of angular made it into the JSON. I have no idea how this happened. I
have searched and searched and I just don't know. Sorry.

Commit:
refactor(karma.conf): add support for all webpack loaders

andrewjensen committed:
Merge pull request #8 from DomoApps/refactor/DAS-4173-cdn-injection

fix(package.json): update version of angular to 1.5.3

Commit:
refactor(package.json,karma.conf): move testing dependencies to package.json

Commit:
refactor(karma.conf.js,package.json): remove testingDeps key and add angular-mocks dep

Commit:
feat(): required store files into common so they will get into bundle. DAS-4096

Commit:
chore(package.json): update da-plop dependency for new testing features

jpumford committed:
Merge pull request #9 from DomoApps/feat/DAS-4146-test-framework-restructuring

Feat/das 4146 test framework restructuring

jpumford committed:
Merge pull request #4 from DomoApps/feat/DAS-4121-framework-versioning

Create framework-specific changelog and contributing docs

Commit:
feat: show 'unsupported' message when trying to authenticate using devtokens

Commit:
refactor(lab.html): remove icon-bits

Commit:
refactor(package.json): replace da-webpack with starter-kit in AppTeam6 config

jpumford committed:
Merge pull request #11 from DomoApps/fix/dev-token-basic-support

feat: show 'unsupported' message when trying to authenticate using devtokens

Commit:
refactor(): Removed the unwanted thumbnail and updated the readme

Commit:
fix(): updated readme with references

Jonathan-Law committed:
Merge pull request #13 from DomoApps/refactor/remove-thumbnail

refacotr(): remove thumbnail from readme and assets

Commit:
Merge remote tracking branch 'origin/master' into feat/DAS-4096-redux-implementation.

Commit:
feat(): change constants file to allow for proper ES6 imports. DAS-4096

Commit:
fix(): corrected the constants examples in constants file. DAS-4096

Commit:
feat(): update da-plop version to include Redux changes. DAS-4096

sldavidson committed:
Merge pull request #5 from DomoApps/feat/DAS-4096-redux-implementation

feat(): Add ngRedux and created common store directories

sldavidson committed:
0.0.6

Commit:
feat(server.es6.js): add support for devtokens

Commit:
refactor(webpack.config): restrict cssnano to only safe minification

jpumford committed:
Merge pull request #44 from DomoApps/refactor/remove-unsafe-minification

refactor(webpack.config): restrict cssnano to only safe minification

Commit:
fix(webpack.config): reveal process.env.NODE_ENV to node modules

because redux needs the node-env to adjust what content it gives us as a module process.env.NODE_ENV
is now exposed. If NODE_ENV is undefined, the process will continue as normal. This squelches a
message we were seeing in the console from redux

jpumford committed:
Merge pull request #45 from DomoApps/fix/DAS-4305-missing-env

fix(webpack.config): reveal process.env.NODE_ENV to node modules

jpumford committed:
v0.0.7

Commit:
fix(switcher): add template with viewport meta tag

include template for switcher's index.html to include a viewport meta tag. The viewport meta tag is

needed for enquire.js to properly work and switch to the correct project

Commit:
fix(switcher): inject switcher.js into body of index.html template

Commit:
perf(webpack.config): enable caching of both sourcemaps and modules

Commit:
style(constants): add extra line at end to satisfy EOL rule

Commit:
chore(lab.html): remove iconbits

Commit:
feat(webpack.config): add support for full sourcemaps while on prod and reduced while on dev

Commit:
fix(babel): switch babel preset to es2015

es2015-loose has some transpiling bugs, for example with sets. Fixes #24

jpumford committed:
Merge pull request #12 from DomoApps/refactor/remove-da-specific-things

Refactor/remove da specific things

Commit:
feat(package.json): added simple support to keep package.json up to date

bmbarker90 committed:
Merge pull request #50 from DomoApps/fix/EOL-in-constants

style(constants): add extra line at end to satisfy EOL rule

bmbarker90 committed:
Merge pull request #53 from DomoApps/fix/switch-babel-preset

fix(babel): switch babel preset to es2015

Commit:
feat: add strict angular dependency injection

jpumford committed:
Merge pull request #46 from DomoApps/fix/DAS-4335-fix-switcher

DAS-4335: fix(switcher): add template with viewport meta tag

jpumford committed:
Merge pull request #43 from DomoApps/fix/dev-token-support

feat(server.es6.js): add support for devtokens

jpumford committed:
Merge pull request #55 from DomoApps/feat/das-4340-ng-strict-di

feat: add strict angular dependency injection

jpumford committed:
Merge pull request #49 from DomoApps/fix/das-4338-slow-builds

perf(webpack.config): enable caching of both sourcemaps and modules

jpumford committed:
v0.0.8

jpumford committed:
Merge pull request #54 from DomoApps/feat/das-4339-manifest-updating

feat(package.json): added simple support to keep package.json up to date

Commit:
feat(webpack.config,eslintrc): add APP_NAME and VERSION to constants

Commit:
fix(package.json): fix `cp -r` command to work properly on linux environments

the `cp -r` command worked inconsistently between osx and linux. cp -r on osx will copy the contents
of the first argument (the source directory) into the output directory. On linux, cp -r would copy
the *source folder* into the destination folder. A blob is used to copy all (except dotfiles) from
the domo folder to the dist folder

jpumford committed:
Merge pull request #62 from DomoApps/fix/domo-97116-fix-build-cp

fix(package.json): fix `cp -r` command to work properly on linux environments

jpumford committed:
Merge pull request #58 from DomoApps/feat/analytics-support

feat(webpack.config,eslintrc): add APP_NAME and VERSION to constants

bmbarker90 committed:
fix(package.json): add clean task (#66)

Commit:
Merge remote-tracking branch 'generator/master'

jpumford commented:
@bmbarker90 it's just an `update-tools`

sldavidson reviewed (changes requested):
Getting an error when trying to run
`ERROR in ./~/angular-material/index.js
Module not found: Error: Cannot resolve module 'angular-animate' in .../advanced-sample-app/node_modules/angular-material
 @ ./~/angular-material/index.js 5:0-26`

I think it just needs to be listed as an external.

Also there is a lot of Redux stubbed out code that isn't being used in this project that we could clean out.

jpumford commented:
@sldavidson Yeah, I thought about the redux stubs! I left them because I was worried removing them would make other `update-tools` pretty complicated, especially if we modify those stubs on `generator`. It's up to you; I can remove the stubs if you'd like.

jpumford commented:
@sldavidson As far as angular-animate goes, should it be added as an external to starter-kit as well, considering it is external? We could run into the same issue

Commit:
fix(package.json): sync angular-aria and angular-mocks with cdnDependencies

Using a newer version of angular-aria with angular-material led to an issue documented here:
https://github.com/angular/material/issues/10475

Commit:
fix(webpack.config): add angular-animate as an external

angular-animate was "missing" as a module since it is loaded from a cdn. This adds it as a webpack
external so angular-material can load the module

jpumford commented:
@sldavidson alright, that's been updated! Let me know what your thoughts are about the redux stubs.

sldavidson commented:
It might confuse people who are looking at the code and see stubbed out code.

sldavidson reviewed (changes requested):
Also looking over the changes, it appears the `cross-evn` from #27 and DomoApps/starter-kit#68 is not getting pulled in.

jpumford commented:
@sldavidson yes, that's because I made the PR when Jake was having issues with cross-env. If you'd like I can add that to this one as well, I just wanted to split up as much as possible (though this PR is still a mess... `update-tools` is going to have to change)
